### PR TITLE
[Thang] seachfragment IU + Logic

### PR DESCRIPTION
**Pull request by huynhthanhthang1296**

### CODE REVIEW

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public void onBindViewHolder(SearchHolder holder, int position) {
+        holder.bind(mMovieList.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return mMovieList.size();
+    }
+
+    public void setOnRecyclerViewItemListener(OnRecyclerViewItemListener onRecyclerViewItemListener) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java]: ERROR: Line is longer than 100 characters (found 102).

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                public void onClick(View view) {
+                    mOnRecyclerViewItemListener.onItemClick(mMovie);
+                }
+            });
+        }
+
+        public void bind(Movie movie) {
+            mMovie = movie;
+            String urlPoster = StringUtils.convertPosterPathToUrlPoster(movie.getPosterPath());
+            String titleMovie = StringUtils.convertLongTitleToShortTitle(movie.getTitle(), movie.getReleaseDate());
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java]: ERROR: Line is longer than 100 characters (found 115).

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchFragment.java [hunk 3]
```diff
@@ -2,19 +2,32 @@
...
+import com.fstyle.androidtrainning.data.service.config.MoviesApi;
 import com.fstyle.androidtrainning.screen.BaseFragment;
+import com.fstyle.androidtrainning.screen.tabhome.OnRecyclerViewItemListener;
+import java.util.List;
 
 /**
  * TabSearch Screen.
  */
-public class TabSearchFragment extends BaseFragment implements TabSearchContract.SearchView {
+public class TabSearchFragment extends BaseFragment implements TabSearchContract.SearchView, OnRecyclerViewItemListener,
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchFragment.java]: ERROR: Line is longer than 100 characters (found 120).

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java [hunk 4]
```diff
@@ -24,4 +33,28 @@ public void onStart() {
...
+    public void setMoviesApi(MoviesApi moviesApi) {
+        mMoviesApi = moviesApi;
+    }
+
+    @Override
+    public void getListSearchMovie(String keyWord) {
+        mMoviesApi.searchMovies(keyWord, Constant.API_KEY, Constant.LANGUAGE,
+                Constant.DEFAULT_PAGE).enqueue(new Callback<GetListMoviesResponse>() {
+            @Override
+            public void onResponse(Call<GetListMoviesResponse> call, Response<GetListMoviesResponse> response) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/TabSearchPresenter.java]: ERROR: Line is longer than 100 characters (found 112).

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    public class SearchHolder extends RecyclerView.ViewHolder {
+
+        private Movie mMovie;
+        private ImageView mPoster;
+        private TextView mMovieName;
+        private OnRecyclerViewItemListener mOnRecyclerViewItemListener;
+        SearchHolder(View itemView, final OnRecyclerViewItemListener onRecyclerViewItemListener) {
+            super(itemView);
+            mPoster = itemView.findViewById(R.id.image_poster_item);
+            mMovieName = itemView.findViewById(R.id.text_name_item);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java]: ERROR: [PMD-CPD] 16 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/MovieAdapter.java

File: app/src/main/res/layout/footer_view.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (daolq3012) [inline app/src/main/res/layout/footer_view.xml]: WARNING: [UnusedResources] The resource `R.layout.footer_view` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+        private Movie mMovie;
+        private ImageView mSearchPoster;
+        private TextView mMovieName;
+        private OnRecyclerViewItemListener mOnRecyclerViewItemListener;
+
+        SearchHolder(View itemView, OnRecyclerViewItemListener onRecyclerViewItemListener) {
+            super(itemView);
+            mSearchPoster = itemView.findViewById(R.id.image_poster_item);
+            mMovieName = itemView.findViewById(R.id.text_name_item);
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/androidtrainning/screen/tabsearch/SearchAdapter.java:76]: ERROR: [PMD-CPD] 16 lines duplicated.
app/src/main/java/com/fstyle/androidtrainning/screen/tabhome/MovieAdapter.java

File: app/src/main/res/layout/fragment_tabsearch.xml [hunk 8]
```diff
@@ -1,18 +1,32 @@
...
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/tool_bar"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:text="@string/search"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:background="#db4c18">
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabsearch.xml]: hard code này em

File: app/src/main/res/layout/fragment_tabsearch.xml [hunk 9]
```diff
@@ -1,18 +1,32 @@
...
+        android:layout_height="wrap_content" />
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recycler_search"
+        android:layout_marginTop="@dimen/dp_10"
+        android:layout_below="@id/tool_bar"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+    </android.support.v7.widget.RecyclerView>
+
```
Comment (tranminhtri911) [inline app/src/main/res/layout/fragment_tabsearch.xml]: remove mấy dòng trống này đi em

**Comment by tranminhtri911** [comment]
ok

**Comment by tranminhtri911** [comment]
Ok

**Pull request merged**